extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php

```diff
@@ -160,4 +149,119 @@ protected function addNode( $node, SimpleXMLElement $xml ) {
 	private function isEmptyNodeValue( $value ) {
 		return ( empty( $value ) && $value != '0' );
 	}
+
+	/**
+	 * @param $type string
+	 * @return string
+	 */
+	protected function getCanonicalType( $type ) {
+		mb_convert_case( $type, MB_CASE_LOWER, 'UTF-8' );
+		$type = !empty($this->typesToCanonicals[$type]) ? $this->typesToCanonicals[$type] : $type;
+		return $type;
+	}
+
+	/**
+	 * @param $infobox
+	 * @return SimpleXMLElement
+	 */
+	protected function createInfoboxXml( $infobox ) {
+		$xml = new SimpleXMLElement( '<' . PortableInfoboxParserTagController::PARSER_TAG_NAME . '/>' );
+		foreach ( $infobox as $key => $value ) {
+			if ( $key !== 'data' ) {
+				$xml->addAttribute( $key, $value );
+			}
+		}
+		$this->addGroupNode( $infobox->data, $xml );
+		return $xml;
+	}
+
+	/**
+	 * @param $xml \SimpleXMLElement
+	 * @param $formatted make the output document human-readable (true) or condensed (false)
+	 * @param $newChild
+	 * @return string
+	 */
+	protected function getFormattedMarkup( $xml, $formatted ) {
+		$infoboxDom = $this->createInfoboxDom( $formatted );
+
+		$inGroup = false;
+		$currentGroupDom = null;
+
+		foreach ( $xml->children() as $currentChildNode ) {
+			$childNodeDom = dom_import_simplexml( $currentChildNode );
+
+			if ( !$inGroup ) {
+				if ( $currentChildNode->getName() !== 'header' ) {
```

Comment:
This is the default, most typical (in a normal infobox) case. This is why I'd like it to be the first one. Any strong counter-opinion ?